src/main/java/org/apache/maven/plugins/checkstyle/AbstractCheckstyleReport.java

```diff
@@ -780,6 +770,53 @@ private void generateMainReport( CheckstyleResults results, ResourceBundle bundl
         generator.generateReport( results );
     }
 
+    private void initializeXrefLocation( CheckstyleReportGenerator generator )
+    {
+      String relativePath = determineRelativePath( xrefLocation );
+      if ( xrefLocation.exists() || checkMavenJxrPluginIsConfigured() )
+      {
+          // XRef was already generated by manual execution of a lifecycle binding
+          // the report is on its way
+          generator.setXrefLocation( relativePath );
+      }
+    }
+
+    private void initializeXrefTestLocation( CheckstyleReportGenerator generator )
+    {
+      String relativePath = determineRelativePath( xrefTestLocation );
+      if ( xrefTestLocation.exists() || checkMavenJxrPluginIsConfigured() )
+      {
+          // XRef was already generated by manual execution of a lifecycle binding
+          // the report is on its way
+          generator.setXrefTestLocation( relativePath );
+      }
+    }
+
+    private String determineRelativePath( File location )
+    {
+      String relativePath = PathTool.getRelativePath( getOutputDirectory(), location.getAbsolutePath() );
+      if ( StringUtils.isEmpty( relativePath ) )
```

Comment:
which StringUtils is this?

Reply:
i use the `StringUtils` class which is already used in the class, hence `org.codehaus.plexus.util.StringUtils`. should we use something different here?

Reply:
Yes, that method likely doesn't do what you think it does and we need to get rid of it.  Possibly use Apache commons or even more clearly don't use a utility method. Looking at the code I think what you want here is "is not null or zero length"  but does allow whitespace. It's a crap shoot whether any given isEmpty method has those semantics or not.  The one you picked doesn't.

Reply:
ok, i get it and agree. i have just used what was already there :-)

the call to PathTool.getRelativePath should actually either return a zero-length string or a valid path. the null check and the whitespaces trimming are just a kind of defensive implementation. we could just look for a zero-length string, e.g. with "relativePath.isEmpty" if that suits you more. 

pushed another commit with explicit evaluation of the string content, including the null check and whitespace trimming.